File: csharp_body_deref/Scripts/CustomLogic/CustomLogicAssignmentExpressionAst.cs
// ==================== AoTTG2 cross-reference ====================
// Type: CustomLogic.CustomLogicAssignmentExpressionAst
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/CustomLogic/CustomLogicAssignmentExpressionAst.c
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicAssignmentExpressionAst.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace CustomLogic;

[Token(Token = "0x2000286")]
internal class CustomLogicAssignmentExpressionAst : CustomLogicBaseExpressionAst
{
	[Token(Token = "0x4000DAD")]
	[FieldOffset(Offset = "0x18")]
	public CustomLogicBaseExpressionAst Left;

	[Token(Token = "0x4000DAE")]
	[FieldOffset(Offset = "0x20")]
	public CustomLogicToken Operator;

	[Token(Token = "0x4000DAF")]
	[FieldOffset(Offset = "0x28")]
	public CustomLogicBaseExpressionAst Right;

	[Token(Token = "0x6000F6E")]
	[Address(RVA = "0x3FB8390", Offset = "0x3FB8390", VA = "0x3FB8390")]
	public CustomLogicAssignmentExpressionAst(CustomLogicBaseExpressionAst left, CustomLogicToken @operator, int line)
	{
	}
}

File: cpp/CustomLogic/CustomLogicAssignmentExpressionAst.c
// Type: CustomLogic.CustomLogicAssignmentExpressionAst
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/CustomLogic/CustomLogicAssignmentExpressionAst.cs
// Prior real C# source (older reference): Assets/Scripts/CustomLogic/Ast/CustomLogicAssignmentExpressionAst.cs
// --------------------------------

// CustomLogic.CustomLogicAssignmentExpressionAst$$.ctor
// il2cpp: void CustomLogic_CustomLogicAssignmentExpressionAst___ctor (CustomLogic_CustomLogicAssignmentExpressionAst_o* __this, CustomLogic_CustomLogicBaseExpressionAst_o* left, CustomLogic_CustomLogicToken_o* operator, int32_t line, const MethodInfo* method);
// 0x40b8390

void CustomLogic_CustomLogicAssignmentExpressionAst___ctor
               (CustomLogic_CustomLogicAssignmentExpressionAst_o *__this,
               CustomLogic_CustomLogicBaseExpressionAst_o *left,CustomLogic_CustomLogicToken_o *operator,
               int32_t line,MethodInfo *method)

{
  System_Object___ctor((Il2CppObject *)__this,(MethodInfo *)0x0);
  (__this->fields).Type = 3;
  (__this->fields).Line = line;
  (__this->fields).Left = left;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Left,left);
  (__this->fields).Operator = operator;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Operator,operator);
  return;
}


